### PR TITLE
add disable option for select and use it in RoomBedDropDown

### LINKED ISSUE #575: possible to select a used bed
### Describe the bug

In the bed selection a bed is shown which is already in use if a bed is selcted that is in use error is occuring

### Describe how to produce the bug

Go to Room Overview Selct a beed that is already in use 

### Additional context

image

### PR DESCRIPTION
**Pull request by DasProffi**
## Which issues does this pull request close?
closes #575

## [OPTIONAL] Give testing instructions to reviewers
Use RoomBedDropDown on PatientDetails

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -10844,3 +10840,7 @@ packages:
 
   /zod@3.21.4:
     resolution: {integrity: sha512-m46AKbrzKVzOzs/DZgVnG5H55N1sv1M8qZU3A8RIKbs3mrACDNeIOeilDymVb2HdmP8uwshOCF4uJ8uM9rCqJw==}
+
+settings:
+  autoInstallPeers: true
+  excludeLinksFromLockfile: false
```
Comment (MaxSchaefer) [inline pnpm-lock.yaml:10846]: Is this configured by us?

File: lib/components/user_input/Select.tsx [hunk 2]
```diff
@@ -47,12 +61,18 @@ export const Select = <T, >({ value, label, options, onChange, isHidingCurrentVa
                 <Menu.Item key={option.label}>
                   {
                     <div
-                      className={tx('px-4 py-2 cursor-pointer overflow-hidden whitespace-nowrap text-ellipsis hover:bg-gray-100 border-2 border-t-0', {
-                        'bg-gray-100 ': option.value === value,
+                      className={tx('px-4 py-2 overflow-hidden whitespace-nowrap text-ellipsis border-2 border-t-0', {
+                        'bg-gray-100': option.value === value,
                         'bg-gray-50': index % 2 === 1,
+                        'text-gray-300 cursor-not-allowed': !!option.disabled,
```
Comment (MaxSchaefer) [inline lib/components/user_input/Select.tsx:67]: The `!!` is not very clean. But I understand that this is in relation with `showDisabledOptions`.

**Review by MaxSchaefer — APPROVED** [review]

File: pnpm-lock.yaml (hunk 1 above)
Comment (DasProffi) [reply pnpm-lock.yaml:10846]: I am unsure it was automatically added, when I did `pnpm install`

File: lib/components/user_input/Select.tsx [hunk 3]
```diff
@@ -47,12 +61,18 @@ export const Select = <T, >({ value, label, options, onChange, isHidingCurrentVa
                 <Menu.Item key={option.label}>
                   {
                     <div
-                      className={tx('px-4 py-2 cursor-pointer overflow-hidden whitespace-nowrap text-ellipsis hover:bg-gray-100 border-2 border-t-0', {
-                        'bg-gray-100 ': option.value === value,
+                      className={tx('px-4 py-2 overflow-hidden whitespace-nowrap text-ellipsis border-2 border-t-0', {
+                        'bg-gray-100': option.value === value,
                         'bg-gray-50': index % 2 === 1,
+                        'text-gray-300 cursor-not-allowed': option.disabled,
```
Comment (MaxSchaefer) [inline lib/components/user_input/Select.tsx:67]:
```suggestion
                        'text-gray-300 cursor-not-allowed': !!option.disabled,
```
Either we require `disabled: boolean` (non null) or casting this value to a boolean.

**Review by MaxSchaefer — DISMISSED** [review]

**Review by MaxSchaefer — APPROVED** [review]

**Pull request merged**